server/src/configuration_set.rs

```diff
@@ -2,13 +2,33 @@
 
 use crate::slice_config;
 use std::path::{Path, PathBuf};
+use std::collections::HashMap;
 use slice_config::SliceConfig;
+use slicec::{ast::Ast, diagnostics::Diagnostic, slice_file::SliceFile};
 use slicec::compilation_state::CompilationState;
 
+#[derive(Debug)]
+pub struct CompilationData {
+    pub ast: Ast,
+    pub files: HashMap<String, SliceFile>,
+}
+
+impl Default for CompilationData {
+    fn default() -> Self {
+        Self {
+            ast: Ast::create(),
+            files: HashMap::default(),
+        }
+    }
+}
+
+unsafe impl Send for CompilationData {}
+unsafe impl Sync for CompilationData {}
```

Comment:
I would be inclined to add a comment above this explaining why we need it and why its safe to do.

Reply:
I added a comment.